### PR TITLE
fix `ipfs files add` CLI command.

### PR DESCRIPTION
**Pull request by noffle**
- fixes 'files add' CLI to work with `createAddStream` changes
- adds tests

### CODE REVIEW

File: src/cli/commands/files/add.js [hunk 1]
```diff
@@ -59,37 +59,39 @@ module.exports = Command.extend({
...
           })
-        } else {
-          rs = fs.createReadStream(inPath)
-          inPath = inPath.substring(inPath.lastIndexOf('/') + 1, inPath.length)
-          filePair = {path: inPath, content: rs}
-          i.write(filePair)
-          i.end()
-        }
+          i.once('end', () => {
+            return
```
Comment (diasdavid) [inline src/cli/commands/files/add.js]: shouldn't it be `process.exit(0)` ?

**Comment by diasdavid** [comment]
Seems that there is a racing condition on `add`

https://travis-ci.org/ipfs/js-ipfs/jobs/135762823#L4512-L4516

```
      Uncaught AssertionError: expected 'QmZ25UfTqXGz9RsEJFg7HUAuBcmfx5dQZDXQd2QEZ8Kj74' to equal 'QmVvjDy7yF7hdnqE8Hrf4MHo5ABDtb5AbX6hWbD3Y42bXP'
      + expected - actual
      -QmZ25UfTqXGz9RsEJFg7HUAuBcmfx5dQZDXQd2QEZ8Kj74
      +QmVvjDy7yF7hdnqE8Hrf4MHo5ABDtb5AbX6hWbD3Y42bXP
```

That come from the tests generated earlier

**Comment by diasdavid** [comment]
Also see https://github.com/ipfs/js-ipfs/pull/306#issuecomment-224016622, related to https://github.com/ipfs/js-ipfs/pull/312#issuecomment-224209639

**Comment by diasdavid** [comment]
Thank you :)

**Comment by noffle** [comment]
Woah, fast merge! I didn't get to squash or address you comments. ;)

**Pull request merged**